src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala

```diff
@@ -162,12 +164,22 @@ final case class PurposeVersionActivation(
     * @param stateChangeDetails Details on the user that is performing the action
     * @return The updated Version
     */
-  def firstVersionActivation(purpose: Purpose, version: PurposeVersion, stateChangeDetails: StateChangeDetails)(implicit
-    contexts: Seq[(String, String)]
-  ): Future[PurposeVersion] = {
+  def firstVersionActivation(
+    purpose: Purpose,
+    version: PurposeVersion,
+    stateChangeDetails: StateChangeDetails,
+    eService: EService
+  )(implicit contexts: Seq[(String, String)]): Future[PurposeVersion] = {
     val documentId: UUID = uuidSupplier.get
     for {
-      path <- createRiskAnalysisDocument(documentId, purpose, version)
+      producer <- partyManagementService.getInstitutionById(eService.producerId)
+      consumer <- partyManagementService.getInstitutionById(purpose.consumerId)
```

Comment:
I'm leaving here a suggestion that is not strictly a review comment to implement. To the best of my understanding, these 2 calls are orthogonal and could be executed in parallel, right? Why don't we perform it like 
```scala
(producer,consumer) <- Future.sequence(
  partyManagementService.getInstitutionById(eService.producerId),
  partyManagementService.getInstitutionById(purpose.consumerId)
)
```
IMHO we should consider adding a few combinators for Future in commons

Reply:
In this particular case, this is totally yak shaving, i do agree @galales :D

Reply:
Local tests and the fact that in these signatures `that` is strict

```scala
def zip[U](that: Future[U]): Future[(T, U)]
def zipWith[U, R](that: Future[U])(f: (T, U) => R)(implicit executor: ExecutionContext): Future[R]
```
makes me think that these futures' execution happens in parallel. I encourage you to test further, though.

Reply:
confirm that `zip` is parallel 👍